analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Branching.LearnConstraint.cs

```diff
@@ -632,6 +632,38 @@ public void Branching_LearnsObjectConstraint_IsNull()
                 .And.ContainSingle(x => x.HasConstraint(ObjectConstraint.NotNull));
         }
 
+        [DataTestMethod]
+        [DataRow("arg.HasValue")]
+        [DataRow("arg.HasValue is true")]
+        [DataRow("arg.HasValue == true")]
+        [DataRow("arg.HasValue != false")]
+        [DataRow("!!arg.HasValue")]
```

Comment:
```suggestion
        [DataRow("!!arg.HasValue")]
        [DataRow("arg.HasValue is not not true")]
```

Reply:
Doens't work, our general pattern matching doesn't support this